cmd/dra-example-kubeletplugin/state.go

```diff
@@ -151,17 +160,65 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 		return nil, fmt.Errorf("claim not yet allocated")
 	}
 
+	// Walk through each device allocation and prepare it.
 	var preparedDevices PreparedDevices
 	for _, result := range claim.Status.Allocation.Devices.Results {
 		if _, exists := s.allocatable[result.Device]; !exists {
 			return nil, fmt.Errorf("requested GPU is not allocatable: %v", result.Device)
 		}
 
-		device := &drapbv1.Device{
-			RequestNames: []string{result.Request},
-			PoolName:     result.Pool,
-			DeviceName:   result.Device,
-			CDIDeviceIDs: s.cdi.GetClaimDevices([]string{result.Device}),
+		// Retrieve the set of device configs for the current allocation.
+		configs, err := GetOpaqueDeviceConfigs(
+			configapi.Decoder,
+			DriverName,
+			result.Request,
+			claim.Status.Allocation.Devices.Config,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error getting GPU configs: %v", err)
+		}
+
+		// Select the GPU config with the highest precedence from those retrieved.
+		// Use the default GPU config if no GPU configs have been returned.
+		config := configapi.DefaultGpuConfig()
+		for _, c := range configs {
+			if c.GetObjectKind().GroupVersionKind() != config.GetObjectKind().GroupVersionKind() {
```

Comment:
If a new type was added to the scheme that is associated with the Decoder, but is not a `GpuConfig` (for example a separate `MigConfig` in the case of NVIDIA GPUs). We don't have another one yet, but we could / maybe should in the future as an example.

Reply:
Actually, wait -- I see your point now. The Decoder will already filter by *some* valid config that is registered with it, and then it is just up to the cast below to pull out the right one. This would also work for other types that are registered as well. 

Will remove the check now.

Reply:
Updated.

Reply:
> The Decoder will already filter by some valid config that is registered with it

Exactly. There is exactly one GroupKind for each type, so if you get that type, it has to match the expected GroupKind.